pkg/validation/attributes/attributes_command.go

```diff
@@ -0,0 +1,108 @@
+package attributes
+
+import (
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	apiAttributes "github.com/devfile/api/v2/pkg/attributes"
+)
+
+// ValidateAndReplaceForCommands validates the commands data for global attribute references and replaces them with the attribute value
+func ValidateAndReplaceForCommands(attributes apiAttributes.Attributes, commands []v1alpha2.Command) error {
+
+	for i := range commands {
+		var err error
+
+		// Validate various command types
+		switch {
+		case commands[i].Exec != nil:
+			if err = validateAndReplaceForExecCommand(attributes, commands[i].Exec); err != nil {
+				return err
+			}
+		case commands[i].Composite != nil:
+			if err = validateAndReplaceForCompositeCommand(attributes, commands[i].Composite); err != nil {
+				return err
+			}
+		case commands[i].Apply != nil:
+			if err = validateAndReplaceForApplyCommand(attributes, commands[i].Apply); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// validateAndReplaceForExecCommand validates the exec command data for global attribute references and replaces them with the attribute value
+func validateAndReplaceForExecCommand(attributes apiAttributes.Attributes, exec *v1alpha2.ExecCommand) error {
+	var err error
+
+	if exec != nil {
+		// Validate exec command line
+		if exec.CommandLine, err = validateAndReplaceDataWithAttribute(exec.CommandLine, attributes); err != nil {
+			return err
+		}
+
+		// Validate exec component
+		if exec.Component, err = validateAndReplaceDataWithAttribute(exec.Component, attributes); err != nil {
+			return err
+		}
```

Comment:
How would this work? `exec.Component` has to be one of the component IDs in the devfile, and so cannot have any `{{}}` chars. This would mean that you could have a global attribute doing something like
```yaml
commands:
  - id: test
    exec:
      component: {{my-component}}
```
but the value of the `my-component` attribute is now bound by the same rules as the value here (must refer to a component key). This sort of thing would also make validation more difficult, as we would have to replace attributes before validating. 

Is there a use case where we would want to configure the target component in this way?

Reply:
makes sense, this was an oversight after your comment regd events in the prev feedback

Reply:
Addressed in new commit :)